ui/angular.json

```diff
@@ -149,6 +165,9 @@
             "stable": {
               "browserTarget": "aou-workbench-ui:build:stable"
             },
+            "preprod": {
```

Comment:
Just thinking out loud, but it seems odd to me that Angular should know anything about which environment it's in.

Reply:
Fundamentally, a build of the UI minimally needs to know what API host to talk to - I don't think there's anything odd about that.

Beyond that, there are practical constants that are complied in, such as the oauth client ID and name of the environment. There are various improvements which could be made as to how these are included, but they are a "build-time" concern at the very least.